docs/redux.md

```diff
@@ -0,0 +1,1228 @@
+# Redux Contributor Guidelines
+
+## 1. **Introduction**
+
+The purpose of this document is to provide our contributors with optimization strategies that they can leverage to implement/refactor our current Redux code to improve our application's performance, and also provide them with Redux best practices when delving into our state storage.
+
+## 2. **Performance Optimization Guidelines**
+## **2.1 Avoiding Expensive Operations in Reducers**
+Emphasize that reducers should be pure and avoid side effects.
+
+```typescript
+// Bad: Performing expensive calculations inside the reducer
+const initialState = { data: [], expensiveResult: 0 };
+
+function myReducer(state = initialState, action) {
+switch (action.type) {
+    case 'ADD_DATA':
+    const newData = action.payload;
+    const expensiveResult = newData.reduce((acc, item) => acc + item.value, 0); // Expensive operation
+    return {
+        ...state,
+        data: [...state.data, newData],
+        expensiveResult,
+    };
+    default:
+        return state;
+    }
+}
+
+// Good: Perform expensive calculations outside the reducer
+const initialState = { data: [], expensiveResult: 0 };
+
+function myReducer(state = initialState, action) {
+    switch (action.type) {
+        case 'ADD_DATA':
+        return {
+            ...state,
+            data: [...state.data, action.payload],
+        };
+        case 'SET_EXPENSIVE_RESULT':
+        return {
+            ...state,
+            expensiveResult: action.payload,
+        };
+        default:
+            return state;
+    }
+}
```

Comment:
C: i like the good/bad examples.
Maybe make that more explicit and visible with this outside of the code block and add emoji?
Suggestion:

🚫 Bad: Performing expensive calculations inside the reducer:
```typescript
const initialState = { data: [], expensiveResult: 0 };
// the bad code example
...
}
```
👍 Good: Perform expensive calculations outside the reducer:
```typescript
const initialState = { data: [], expensiveResult: 0 };
// the good code example
...
}
```